examples/lstm_model.py

```diff
@@ -0,0 +1,131 @@
+        # and the output from the last timestep
+        nonlocal t_counter, x_counter
+        z = m.output(x_counter)
+        z = m2.InputContainer(z.matrix)
+        x_counter = m.next_state(x_counter, future_loading(t), t - t_counter)
+        t_counter = t
+        return z
+    
+    results2 = m2.simulate_to(data.times[-1], future_loading2, dt=TIMESTEP, save_freq=TIMESTEP)
+
+    # Step 4: Compare model to original model
+    print('Comparing results...')
+    data.outputs.plot(title='original model')
+    results2.outputs.plot(title='generated model')
+    plt.show()
+
+    # -----------------------------------------------------
+    # Example 2- variable timestep 
+    # Here we will create a model to work with any timestep
+    # We do this by adding timestep as a variable in the model
+    # -----------------------------------------------------
+
+    # Step 1: Generate additional data
+    # We will use data generated above, but we also want data at additional timesteps 
+    print('\n------------------------------------------\nExample 2...')
+    print('Generating additional data...')
+    data_half = m.simulate_to_threshold(future_loading, threshold_keys='impact', save_freq=TIMESTEP/2, dt=TIMESTEP/2)
+    data_quarter = m.simulate_to_threshold(future_loading, threshold_keys='impact', save_freq=TIMESTEP/4, dt=TIMESTEP/4)
+    data_twice = m.simulate_to_threshold(future_loading, threshold_keys='impact', save_freq=TIMESTEP*2, dt=TIMESTEP*2)
+    data_four = m.simulate_to_threshold(future_loading, threshold_keys='impact', save_freq=TIMESTEP*4, dt=TIMESTEP*4)
+
+    # Step 2: Data Prep
+    # We need to add the timestep as a input
+    u = np.array([[TIMESTEP] for _ in data.inputs])
+    u_half = np.array([[TIMESTEP/2] for _ in data_half.inputs])
+    u_quarter = np.array([[TIMESTEP/4] for _ in data_quarter.inputs])
+    u_twice = np.array([[TIMESTEP*2] for _ in data_twice.inputs])
+    u_four = np.array([[TIMESTEP*4] for _ in data_four.inputs])
+
+    training_data = [
+        (u, data.outputs),
+        (u_half, data_half.outputs),
+        (u_quarter, data_quarter.outputs),
+        (u_twice, data_twice.outputs),
+        (u_four, data_four.outputs)
+    ]
+
+    # Step 3: Generate Model
+    print('Building model...')
+    m3 = LSTMStateTransitionModel.from_data(
+        training_data,  
+        sequence_length=4, 
+        epochs=40, 
+        inputs = ['dt'],
+        outputs = ['x'])    
+
+    # Step 4: Simulate with model
+    t_counter = 0
+    x_counter = m.initialize()
+    def future_loading3(t, x = None):
+        nonlocal t_counter, x_counter
+        z = m3.InputContainer({'x_t-1': x_counter['x'], 'dt': t - t_counter})
+        x_counter = m.next_state(x_counter, future_loading(t), t - t_counter)
+        t_counter = t
+        return z
+
+    # Use new dt, not used in training
+    data = m.simulate_to(data.times[-1], future_loading, dt=TIMESTEP*3, save_freq=TIMESTEP*3)
```

Comment:
Why is this not used in training?

Reply:
added a comments. Does the comment above answer your question?

Reply:
Yep makes more sense now!